### ISSUE TITLE
Empty but visible images consume high CPU on 2.4.x

### ISSUE THREAD

**Issue by dweymouth** [fyne-io/fyne#4345]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

A canvas.Image that is empty (no image.Image, fyne.Resource or file set) but not hidden will cause high CPU use as it causes constant texture cache misses and a new GL texture to be created for the "image" on every redraw. This is a regression from 2.3.x.

### How to reproduce

Run the example app

### Example code

```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/canvas"
)

func main() {

	myApp := app.New()
	myWindow := myApp.NewWindow("Empty image perf regression")

	myWindow.SetContent(canvas.NewImageFromImage(nil))
	myWindow.Resize(fyne.NewSize(400, 400))
	myWindow.ShowAndRun()
}
```

### Fyne version

2.4.1

### Go compiler version

go1.21.1

### Operating system and version

macOS Ventura M2

**Comment by dweymouth** [comment]
OK I've managed to work out that it's texture cache misses. newGlImageTexture which is in the pprof trace is only referenced [here](https://github.com/fyne-io/fyne/blob/ccc6a1c19a6d16988da3e468f08f25d8100cd4b4/internal/painter/gl/draw.go#L38) in drawImage, and print statements in [getTexture](https://github.com/fyne-io/fyne/blob/master/internal/painter/gl/texture.go#L32) (which is called by drawTextureWithDetails) reveal that on 2.3.x there are a handful of cache misses on startup (as expected), but then none until mousing over an image. On 2.4.x however it continues to have dozens of cache misses a second even when not interacting with the app

**Comment by andydotxyz** [comment]
Should we mark this as blocker for v2.4.3 (i.e. the point release after current work)?

**Comment by dweymouth** [comment]
> Should we mark this as blocker for v2.4.3 (i.e. the point release after current work)?

I imagine this should be a quick fix to someone familiar with the image painter but I don't know what the fix is. On the other hand it has a very easy workaround (hide empty images) so personally I don't feel it needs to be high priority, unless others complain about it.

**Comment by andydotxyz** [comment]
I'm inclined to agree - unless there is a common case when images may be empty and developers won't know, or be able to tell?

**Comment by andydotxyz** [comment]
I wonder if the following is a good solution? It doesn't seem to take the CPU usage to 0 though...

```go
--- a/internal/painter/gl/draw.go
+++ b/internal/painter/gl/draw.go
@@ -36,6 +36,9 @@ func (p *painter) drawGradient(o fyne.CanvasObject, texCreator func(fyne.CanvasO
 }
 
 func (p *painter) drawImage(img *canvas.Image, pos fyne.Position, frame fyne.Size) {
+       if img.Image == nil && img.File == "" && img.Resource == nil || img.Translucency == 1.0 {
+               return
+       }
        p.drawTextureWithDetails(img, p.newGlImageTexture, pos, img.Size(), frame, img.FillMode, float32(img.Alpha()), 0)
 }
```

**Comment by andydotxyz** [comment]
I wonder if the following is a good solution? It doesn't seem to take the CPU usage to 0 though...

```go
--- a/internal/painter/gl/draw.go
+++ b/internal/painter/gl/draw.go
@@ -36,6 +36,9 @@ func (p *painter) drawGradient(o fyne.CanvasObject, texCreator func(fyne.CanvasO
 }
 
 func (p *painter) drawImage(img *canvas.Image, pos fyne.Position, frame fyne.Size) {
+       if img.Image == nil && img.File == "" && img.Resource == nil || img.Translucency == 1.0 {
+               return
+       }
        p.drawTextureWithDetails(img, p.newGlImageTexture, pos, img.Size(), frame, img.FillMode, float32(img.Alpha()), 0)
 }
```

**Comment by andydotxyz** [comment]
Resolved by #5290, I'm glad to have found a better solution than I proposed before - it helps when you identify the root cause ;)
